src/dist/replication/lib/replica_split.cpp

```diff
@@ -474,7 +474,134 @@ void replica::child_catch_up_states() // on child partition
+                   get_ballot());
+
+    rpc::call(_config.primary,
+              RPC_SPLIT_NOTIFY_CATCH_UP,
+              *request,
+              tracker(),
+              [this](error_code ec, notify_cacth_up_response &&resp) {
+                  auto response = std::move(resp);
+                  if (ec == ERR_TIMEOUT) {
+                      dwarn_replica("notify primary catch up timeout, please wait and retry");
+                      tasking::enqueue(LPC_PARTITION_SPLIT,
+                                       tracker(),
+                                       std::bind(&replica::child_notify_catch_up, this),
+                                       get_gpid().thread_hash(),
+                                       std::chrono::seconds(1));
+                      return;
+                  }
+                  if (ec != ERR_OK || response.err != ERR_OK) {
+                      error_code err = (ec == ERR_OK) ? response.err : ec;
+                      dwarn_replica("failed to notify primary catch up, error={}", err.to_string());
+                      _stub->split_replica_error_handler(
+                          _split_states.parent_gpid,
+                          std::bind(&replica::parent_cleanup_split_context, std::placeholders::_1));
+                      child_handle_split_error("notify_primary_split_catch_up");
+                      return;
+                  }
+                  ddebug_replica("notify primary catch up succeed");
+              },
+              std::chrono::seconds(0),
+              _split_states.parent_gpid.thread_hash());
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica::parent_handle_child_catch_up(const notify_catch_up_request &request,
+                                           notify_cacth_up_response &response) // on primary parent
+{
+    if (status() != partition_status::PS_PRIMARY) {
+        derror_replica("status is {}", enum_to_string(status()));
+        response.err = ERR_INVALID_STATE;
+        return;
+    }
+
+    if (request.child_ballot != get_ballot()) {
+        derror_replica("receive out-date request, request ballot = {}, local ballot = {}",
+                       request.child_ballot,
+                       get_ballot());
+        response.err = ERR_INVALID_STATE;
+        return;
+    }
+
+    if (request.child_gpid != _child_gpid) {
+        derror_replica(
+            "receive wrong child request, request child_gpid = {}, local child_gpid = {}",
+            request.child_gpid,
+            _child_gpid);
+        response.err = ERR_INVALID_STATE;
+        return;
+    }
+
+    response.err = ERR_OK;
+    ddebug_replica("receive catch_up request from {}@{}, current ballot={}",
+                   request.child_gpid,
+                   request.child_address.to_string(),
+                   request.child_ballot);
+
+    _primary_states.caught_up_child.insert(request.child_address);
+    for (auto &iter : _primary_states.statuses) {
```

Comment:
Add a comment on `node_statuses statuses;` of `primary_context`, I'm not clear what it represent for.